conans/test/integration/toolchains/gnu/test_gnutoolchain.py

```diff
@@ -0,0 +1,437 @@
+                 "profile_build": profile_build,
+                 "profile_host": profile_host})
+    client.run("install . --profile:build=profile_build --profile:host=profile_host")
+    conanbuild = client.load("conanbuild.conf")
+    host_flags = re.findall(r"--host=[\w-]*\b", conanbuild)
+    build_flags = re.findall(r"--build=[\w-]*\b", conanbuild)
+    assert len(host_flags) == 1
+    assert len(build_flags) == 1
+    assert host_flags[0] == '--host=x86_64-my-triplet-host'
+    assert build_flags[0] == '--build=arm-my-triplet-build'
+
+
+def test_msvc_profile_defaults():
+    """
+    Tests how GnuToolchain manages MSVC profile and its default env variables.
+    """
+    profile = textwrap.dedent("""\
+    [settings]
+    os=Windows
+    arch=x86_64
+    compiler=msvc
+    compiler.version=191
+    compiler.runtime=dynamic
+    build_type=Release
+
+    [conf]
+    tools.build:compiler_executables={"c": "clang", "cpp": "clang++"}
+    # Fake installation path
+    tools.microsoft.msbuild:installation_path={{os.getcwd()}}
+    """)
+    # Consumer with default values
+    consumer = textwrap.dedent("""\
+    import os
+    from conan import ConanFile
+    from conan.tools.files import load
+
+    class consumerRecipe(ConanFile):
+        name = "consumer"
+        version = "1.0"
+        # Binary configuration
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "GnuToolchain"
+
+        def build(self):
+            toolchain = os.path.join(self.generators_folder, "conanautotoolstoolchain.bat")
+            content = load(self, toolchain)
+            # Default values and conf ones
+            assert r'set "CC=clang"' in content  # conf value has precedence
+            assert r'set "CXX=clang++"' in content  # conf value has precedence
+            assert 'set "LD=link -nologo"' in content
+            assert r'set "AR=lib"' in content
+            assert 'set "NM=dumpbin -symbols"' in content
+            assert 'set "OBJDUMP=:"' in content
+            assert 'set "RANLIB=:"' in content
+            assert 'set "STRIP=:"' in content
+    """)
+    client = TestClient()
+    client.save({
+        "profile": profile,
+        "consumer/conanfile.py": consumer
+    })
+    client.run("create consumer -pr:a profile --build=missing")
+    # Consumer changing default values
+    consumer = textwrap.dedent("""\
+    import os
+    from conan import ConanFile
+    from conan.tools.files import load
+    from conan.tools.gnu import GnuToolchain
+
+    class consumerRecipe(ConanFile):
+        name = "consumer"
+        version = "1.0"
+        # Binary configuration
+        settings = "os", "compiler", "build_type", "arch"
+
+        def generate(self):
+            tc = GnuToolchain(self)
+            # Prepending compiler wrappers
+            tc.extra_env.prepend("CC", "compile")
+            tc.extra_env.prepend("CXX", "compile")
+            tc.extra_env.prepend("AR", "ar-lib")
```

Comment:
@jcar87 I removed the conf `tools.build:compiler_wrappers` because of the downsides. Anyway, as you can see it's quite easy to add those.

For instance, the `bison` recipe could look like the following example:

```python
    def generate(self):
        env = VirtualBuildEnv(self)
        env.generate()
        tc = GnuToolchain(self)
        extra_configure_args = {
            "--enable-relocatable": None,
            "--disable-nls": None,
            "--datarootdir": "${prefix}/res",
        }
        if self.settings.compiler == "apple-clang":
            extra_configure_args["gl_cv_compiler_check_decl_option"] = " "
        if is_msvc(self):
            compile_wrapper = unix_path(self, self.conf.get("user.automake:compile-wrapper", check_type=str))
            lib_wrapper = unix_path(self, self.conf.get("user.automake:lib-wrapper", check_type=str))
            tc.extra_env.prepend("CC", compile_wrapper)
            tc.extra_env.prepend("CXX", compile_wrapper)
            tc.extra_env.prepend("AR", lib_wrapper)
            # Avoid a `Assertion Failed Dialog Box` during configure with build_type=Debug
            # Visual Studio does not support the %n format flag:
            # https://docs.microsoft.com/en-us/cpp/c-runtime-library/format-specification-syntax-printf-and-wprintf-functions
            # Because the %n format is inherently insecure, it is disabled by default. If %n is encountered in a format string,
            # the invalid parameter handler is invoked, as described in Parameter Validation. To enable %n support, see _set_printf_count_output.
            extra_configure_args.update({
                "gl_cv_func_printf_directive_n": "no",
                "gl_cv_func_snprintf_directive_n": "no",
            })
        tc.configure_args.update(extra_configure_args)
        tc.generate()
```